environs/manual/linux/provisioner.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2016 Canonical Ltd.
+// Copyright 2016 Cloudbase Solutions SRL
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package linux
+
+import (
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/environs/manual/common"
+
+	"github.com/juju/loggo"
+)
+
+var (
+	logger = loggo.GetLogger("juju.environs.manual.linux")
+)
+
+const Scope = "ssh"
+
+type provision struct {
+}
+
+// NewProvisioner returns a new linux provision
+func NewProvisioner() *provision {
+	return &provision{}
+}
+
+// Provision returns a new machineId and nil if the provision process is done successfully
+// The func will manual provision a linux machine using as it's default protocol SSH
+func (p provision) ProvisionMachine(args common.ProvisionMachineArgs) (machineId string, err error) {
+	defer func() {
```

Comment:
This isn't something that should go in a `defer` statement. This should be normal handled in the normal error-path of this method.